cmake/setup_tests.cmake

```
@@ -50,6 +50,37 @@ else()
   message(STATUS "Unit tests with GoogleTest: disabled")
 endif()
 
+# Fetch Google Benchmark and setup benchmark tests if option is enabled
+four_c_process_global_option(
+  FOUR_C_WITH_GOOGLE_BENCHMARK "Use Google Benchmark for micro benchmark tests" OFF
+  )
+if(FOUR_C_WITH_GOOGLE_BENCHMARK)
+  # Define a convenience target for all benchmark tests and add it to 'full'
+  # All benchmark test executables should add themselves as a dependency to 'benchmarktests'
+  # disable tests from google benchmark
+  set(BENCHMARK_ENABLE_TESTING OFF)
+  add_custom_target(benchmarktests)
+  add_dependencies(full benchmarktests)
+
+  if(TARGET benchmark_main)
+    message(
+      FATAL_ERROR "A target <benchmark_main> has already been included by a TPL."
```

```suggestion
      FATAL_ERROR "A target <benchmark_main> has already been included by another library."
```

I tried to avoid the word "TPL" in #595.